probe-rs/src/flashing/erase.rs

```diff
@@ -65,13 +65,11 @@ pub fn erase_all(session: &mut Session, progress: Option<FlashProgress>) -> Resu
                     regions.iter().any(|r| r.range.contains_range(&range))
                 })
                 .collect::<Vec<_>>();
-            let qty = sectors.len();
             let first = sectors.first().unwrap();
+            let last = sectors.last().unwrap();
             flasher.run_erase(|active| {
-                active.erase_sectors(
-                    first.base_address.try_into().unwrap(),
-                    qty.try_into().unwrap(),
-                )?;
+                active.erase_range(first.base_address, last.base_address + last.size - 1)?;
```

Comment:
How come you went with inclusive? If I'm not mixing up my intervals, the end of an exclusive range is just `address + size` 🤔

Reply:
My brain malfunctioning, basically :facepalm:.
I probably should have not done any programming today.